### DISCUSSION TITLE
Trying to build a Verify integration, but not sure where to start

### DISCUSSION THREAD

**Discussion by nth-commit** [VerifyTests/Verify#598] [Q&A]
I've been building a property-based testing framework and I've been using some of the primitives it exposes to add an easy way to take snapshots from random data.

For example;

```csharp
using GalaxyCheck;

namespace IntegrationSample
{
    public class GenSnapshotTests
    {
        public record GeneratedInput(int A, int B, int C);

        public record TestOutput(int Sum);

        [GenSnapshot]
        public TestOutput Addition(GeneratedInput input)
        {
            var result = new TestOuput(input.A + input.B + input.C);

            return result;
        }
    }
}
```

- `GenSnapshotAttribute` inherits from `Xunit.Fact`
- The test method is given random (but deterministic) input data
- The test method returns some output, which I want to snapshot

To do the snapshotting, you need to register some global configuration. This example shows how I was able to integrate with Snapshooter:

```csharp
using GalaxyCheck.Configuration;
using Snapshooter;
using Snapshooter.Xunit;
using System.IO;
using System.Threading.Tasks;

namespace IntegrationSample
{
    internal class ConfigureGalaxyCheck : IConfigureGlobal
    {
        public void Configure(IGlobalConfiguration config)
        {
            config.GenSnapshots.AssertSnapshotMatches = (snapshot) =>
            {
                var snapshotName = $"{snapshot.TestClassName}.{snapshot.TestMethodName}_{snapshot.Seed}.snap";
                var directoryName = Path.GetDirectoryName(snapshot.TestFileName)!;
                Snapshot.Match(snapshot.Output, new SnapshotFullName(snapshotName, directoryName));
                return Task.CompletedTask;
            };
        }
    }
}
```

- You register your snapshot tool in the global config, on the delegate `Func<ISnapshot, Task>`
- The function is responsible for executing the snapshot
- The snapshot structure given to the delegate has file information on it (`GenSnapshotAttribute` captures the filename of the class)
- For reference, the `ISnapshot` structure is defined here: https://github.com/nth-commit/GalaxyCheck/blob/a798c9e7fbdecbbc96eaa24d5974a299c4e4849a/src/GalaxyCheck.Xunit/Configuration/IGlobalConfiguration.cs#L29-L42

**Is it possible to do something like this with Verify?**

I've been looking at how the other integrations are building (e.g. Verify.Xunit), and it seems they are dependent on `InnerVerify`, which is an internal to the package, and not accessible by a third-party.

**Comment by nth-commit** [comment]
Bump 🙏

**Reply by SimonCropp** [reply]
InnerVerify is now public, does that unblock you?

**Comment by SimonCropp** [comment]
the closes thing to AssertSnapshotMatches  is comparers https://github.com/VerifyTests/Verify/blob/main/docs/comparer.md
